cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -2262,8 +2262,7 @@ TEST(Cast, StructToSameSizedButDifferentNamedStruct) {
 
   EXPECT_RAISES_WITH_MESSAGE_THAT(
       TypeError,
-      ::testing::HasSubstr("Type error: struct field names do not match: struct<a: int8, "
-                           "b: int8> struct<c: int8, d: int8>"),
+      ::testing::HasSubstr("struct (sub)fields don't match or are in the wrong order"),
```

Comment:
To make the tests pass, because this kind of extends the already implemented `CastStruct`.
(Ditto below and in `dataset/scanner_test.cc`.

Reply:
This makes me think, should we just extend the already existing `CastStruct` implementation?